src/tools/mesh_tester.cpp

```diff
@@ -250,6 +328,97 @@ std::vector< std::pair<int, int> > naiveIntersectionAlgorithm(
   return retval;
 }
 
+#if defined(AXOM_USE_RAJA)
+template < typename ExecSpace >
+std::vector< std::pair<int, int> > naiveIntersectionAlgorithm(
+  mint::Mesh* surface_mesh,
+  std::vector<int> & degenerate)
+{
+  SLIC_INFO("Running naive intersection algorithm "
+    << " in execution Space: " 
+    << axom::execution_space< ExecSpace >::name());
+
+  // Get allocator
+  int allocatorID = axom::execution_space< ExecSpace >::allocatorID();
+  axom::setDefaultAllocator( allocatorID );
+
+  std::vector< std::pair<int, int> > retval;
+
+  const int ncells = surface_mesh->getNumberOfCells();
+  SLIC_INFO("Checking mesh with a total of "<< ncells<< " cells.");
+
+  Triangle3 * tris = axom::allocate <Triangle3> (ncells);
+
+  // Get each triangle in the mesh and check for degeneracies
+  for (int i = 0; i < ncells ; i++)
+  {
+    tris[i] = getMeshTriangle(i, surface_mesh);
+    if (tris[i].degenerate())
+    {
+      degenerate.push_back(i);
+    }
+  }
+
+  RAJA::RangeSegment row_range(0, ncells);
+  RAJA::RangeSegment col_range(0, ncells);
+
+  using KERNEL_POL = 
+    typename axom::mint::internal::structured_exec< ExecSpace >::loop2d_policy;
+  using REDUCE_POL =
+    typename axom::execution_space< ExecSpace >::reduce_policy;
+  using ATOMIC_POL = 
+    typename axom::execution_space< ExecSpace >::atomic_policy;
+ 
+  RAJA::ReduceSum< REDUCE_POL, int > numIntersect(0);  
+
+  // Compute the number of intersections
+  RAJA::kernel<KERNEL_POL>( RAJA::make_tuple(col_range, row_range),
+    AXOM_LAMBDA(int col, int row) {
+    if (row > col)
+    {
+      if (checkTT (tris[row], tris[col]))
+      {
+        numIntersect += 1;
+      }
+    }
+  });
+
+  // Allocation to hold intersection pairs and counter to know where to store
+  int * intersections =
+    axom::allocate <int> (numIntersect.get() * 2);
+  int * counter = axom::allocate <int> (1);
+  
+  counter[0] = 0;
+
+  // RAJA loop to populate with intersections
+  RAJA::kernel<KERNEL_POL>( RAJA::make_tuple(col_range, row_range),
+    AXOM_LAMBDA(int col, int row) {
+    if (row > col)
+    {
+      if (checkTT (tris[row], tris[col]))
+      {
+        auto idx = RAJA::atomicAdd<ATOMIC_POL>(counter, 2);
+        intersections[idx] = row;
+        intersections[idx+1] = col;
+      }
+    }
+  });
+
+  // Initialize pairs of clashes
+  for (auto i = 0 ; i < numIntersect.get() * 2 ; i += 2)
```

Comment:
I would have hoisted out the `int N = numIntersect.get()*2;` into a variable. The compiler should do that and not do the multiplication at every iteration, but, I tend to do that anyways...